jackrabbit-core/src/main/java/org/apache/jackrabbit/core/security/user/UserManagerImpl.java

```diff
@@ -911,6 +933,13 @@ private Authorizable internalGetAuthorizable(String id) throws RepositoryExcepti
         return getAuthorizable(n);
     }
 
+    private <T> T getAuthorizableByType(Authorizable authorizable, Class<T> authorizableClass) throws AuthorizableTypeException {
+        if (authorizable == null || authorizableClass.isInstance(authorizable)) {
+            return authorizableClass.cast(authorizable);
+        }
+        throw new AuthorizableTypeException("Invalid authorizable type '" + authorizable.getClass() + "'");
```

Comment:
I think the null check was not required since both Class#isInstance and Class#cast ingest null and document the behavior.

However, if authorizableClass is null, there will be NPE, but I think that's expected. I have updated the documentation to require non null authorizableClass parameter.

Changed the PR to catch the CCE.